### PR TITLE
Fix conditional order of precedence

### LINKED ISSUE #1955: Conditional order of precedence bug in ScrollableControl.SetDisplayRectLocation
From the code in https://github.com/dotnet/winforms/blob/3f564803ff268bab776fea7df9970d37760ed4da/src/System.Windows.Forms/src/System/Windows/Forms/ScrollableControl.cs#L786-L798

The conditional check `if (xDelta != 0 || yDelta != 0 && IsHandleCreated)` has incorrect bracketing and order of precedence.

It should be `if ((xDelta != 0 || yDelta != 0) && IsHandleCreated)`.

This has observable differences in behaviour - see #1920 (https://dotnetfiddle.net/EVSyDY)

Fixing this bug would have the following implications
- We don't create a `Handle` if `xDelta != 0` and `IsHandleCreated == true`

### PR DESCRIPTION
**Pull request by RussKie**
## Proposed changes

- fix warning SA1408: Conditional expressions should declare precedence
As per docs `&&` operator has precedence over `||` and thus the following changes are to increase readability and make SCA happy. The code was correct, it could not have been interpreted any different. 
- `ScrollableControl.SetDisplayRectLocation` had an incorrect order of precedence by which a `ScrollWindowEx` message could get sent to the control even though the control did not have a valid handle.
Fixes #1955
- `COM2PropertyDescriptor.SetValue` had an incorrect order of precedence that would lead to IOORE if error info string was empty.
Extract the code in question in a standalone method so it could be tested. After the tests, the implementation is rewritten to reduce allocations.
An alternative and significantly shorter implementation is considered: `strMessage.ToString().TrimEnd('\r', '\n')` however uses significantly more memory, though it is slightly more performant.
    ```
    |      Method |       Mean |      Error |     StdDev | Ratio |  Gen 0 |  Gen 1 | Gen 2 | Allocated |
    |------------ |-----------:|-----------:|-----------:|------:|-------:|-------:|------:|----------:|
    |    Original | 1,923.3 ns | 13.6586 ns | 12.1080 ns |  1.00 | 3.0899 | 0.0076 |     - |   19384 B |
    |    Proposed |   207.7 ns |  0.8548 ns |  0.7995 ns |  0.11 | 0.0725 |      - |     - |     456 B |
    |    Alt trim |   165.8 ns |  1.3424 ns |  1.1900 ns |  0.09 | 0.1631 | 0.0005 |     - |    1024 B |
    ```
*  `ComboBox.WndProc` appears to may have handled `WM_PRINTCLIENT` incorrectly in cases a combo had `FlatStyle.Popup` but with `UserPaint` style applied.
Fix the logic by handling `WM_PRINTCLIENT` message only if 
    * the control is painted by the system, and
    * the control has flat style set to `Flat` or `Popup`
* `SplitContainer.OnKeyDown`

## Customer Impact

- We did not have any reported issues directly attributable to the updated code. 
The issue #1955 was found by @hughbe while developing unit tests. 
Other cases were discovered after [SCA SA1408 rule](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1408.md) was enabled.

## Regression? 

- No, it is likely the code had issues for a long time

## Risk

- Consumers who may have developed workarounds for the issues discovered may now be getting unexpected results.

## Test methodology 

- manual
- CTI

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/1985)

### CODE REVIEW

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ScrollableControlTests.cs [hunk 1]
```diff
@@ -743,6 +796,8 @@ public new bool VScroll
 
             public new void AdjustFormScrollbars(bool displayScrollbars) => base.AdjustFormScrollbars(displayScrollbars);
 
+            public void ForceCreateHandle() => base.CreateHandle();
```
Comment (hughbe) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ScrollableControlTests.cs]: Nit: usually in the tests we just write `Assert.NotEqual(IntPtr.Zero, control.Handle)` but this isn't vital

Comment (RussKie) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ScrollableControlTests.cs]: Thank you for the review.
This method is not for assertions, I need to force a handle creation to ensure the control has scrollbars rendered, and thus has correctly calculated `ClientRectangle`.

Comment (RussKie) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ScrollableControlTests.cs]: I had another look at the code and you comment now makes sense - by invoking `child.Handle` property a handle gets created, if there is no valid handle available.

**Comment by RussKie** [comment]
Recommend reviewing commit by commit

**Comment by zsd4yr** [comment]
@merriemcgaw also flagging as expected 3.1 work for Lee's query

**Comment by merriemcgaw** [comment]
Let's put this in 5.0

**Comment by RussKie** [comment]
Rebased on the master

**Pull request merged**